packages/flutter_driver/lib/src/common/diagnostics_tree.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'enum_util.dart';
+import 'find.dart';
+import 'message.dart';
+
+/// [DiagnosticsNode] tree types that can be requested by [GetDiagnosticsTree].
+enum DiagnosticsType {
+  /// The [DiagnosticsNode] tree formed by [RenderObject]s.
+  renderObject,
+
+  /// The [DiagnosticsNode] tree formed by [Widget]s.
+  widget,
+}
+
+EnumIndex<DiagnosticsType> _diagnosticsTypeIndex = EnumIndex<DiagnosticsType>(DiagnosticsType.values);
+
+/// A Flutter Driver command to retrieve the json-serialized [DiagnosticsNode]
+/// tree of the object identified by [finder].
+///
+/// The [DiagnosticsType] of the [DiagnosticsNode] tree returned is specified by
+/// [diagnosticsType].
+class GetDiagnosticsTree extends CommandWithTarget {
+  /// Creates a [GetDiagnosticsTree] Flutter Driver command.
+  GetDiagnosticsTree(SerializableFinder finder, this.diagnosticsType, {
+    this.subtreeDepth = 0,
+    this.includeProperties = true,
+    Duration timeout,
+  }) : super(finder, timeout: timeout);
+
+  /// Deserializes this command from the value generated by [serialize].
+  GetDiagnosticsTree.deserialize(Map<String, dynamic> json)
+      : subtreeDepth = json['subtreeDepth'] == 'null' ? null : int.parse(json['subtreeDepth']),
+        includeProperties = json['includeProperties'] == 'true',
+        diagnosticsType = _diagnosticsTypeIndex.lookupBySimpleName(json['diagnosticsType']),
+        super.deserialize(json);
+
+  /// How many levels of children to include in the JSON result.
+  ///
+  /// Defaults to zero, which will only return the [DiagnosticsNode] information
+  /// of the object identified by [finder].
+  final int subtreeDepth;
+
+  /// Whether the properties of a [DiagnosticsNode] should be included.
+  final bool includeProperties;
+
+  /// The type of [DiagnosticsNode] tree that is requested.
+  final DiagnosticsType diagnosticsType;
+
+  @override
+  Map<String, String> serialize() => super.serialize()..addAll(<String, String>{
+    'subtreeDepth': subtreeDepth.toString(),
```

Comment:
NPE if `subtreeDepth` is null

Reply:
Calling toString on null is actually perfectly legal: https://dartpad.dartlang.org/be63aa1861c7e7c1dccfbd159121dba0 :)